src/topic_training/src/Demo.cpp

```diff
@@ -0,0 +1,136 @@
+/* Include all headers of the ROS system we are using 
+   separately. */
+#include "ros/init.h"
+#include "ros/node_handle.h"
+#include "ros/rate.h"
+#include "ros/subscriber.h"
+#include "ros/publisher.h"
+
+/* Include the message type we will be publishing. */
+#include "std_msgs/String.h"
+
+/* Include Stringstream to be able to manipulate the 
+   data we are sending and receiving. (Optional: only 
+   if the message is of type String)*/
+#include <sstream>
+
+/* C++ does not like "magic numbers," so it is better 
+   if we make them variables. */
```

Comment:
This isn't just a C++ thing.  Imagine you're reading some java and somebody keeps throwing the number 3735928559 around.  It's a common cryptographic initialization value, but it doesn't matter what language you're using; that one's not easy to figure out.
```suggestion
/* Avoid unexplained literal numbers in code */
```

Reply:
Fixed in ce332e7.